src/annotated_lcf.fun

```diff
@@ -0,0 +1,40 @@
+functor AnnotatedLcf
```

Comment:
Some interesting things I learnt while making this code compatible with MLTon.

1. SML/NJ's support for "higher order functors" is nonstandard; you resolve this by currying the inputs into a single structure.

2. SML/NJ's where-clauses for structures are nonstandard. You can resolve this by using structure sharing, but this works differently in MLTon from in SML/NJ: `sharing X = Y` induces sharing constraints for all the type components of `X`, but in SML/NJ it excludes the types that have manifest definitions, whereas in MLTon, it does not. Because you cannot express a sharing constraint on a type that has a manifest definition, this results in an error in many cases. I believe that MLTon's behavior is the correct one, even though it is inconvenient. The solution is to just use the transparent ascription; to avoid exposing types that we want to remain abstract (like `tactic`), we go through an intermediate structure `Private` where we enforce the correct level of opacity.